content/BCPatterns/generic-method-pattern/index.md

```diff
@@ -0,0 +1,327 @@
++++
+title = "Generic Method Pattern"
+weight = 1180
++++
+
+_Created by Gary Winter (Cloud Ready Software), Described by waldo (iFacto Business Solutions | Dynex)_
+
+## Abstract
+
+The goal of this pattern is to facilitate a lot of things in one single awesome way of writing code.  If you apply this apply this pattern as a general pattern, you'll implement:
+- Extensibility
+- Decouplability
+- Readability
+- Testability
+- Encapsulation
+
+## Context: 
+
+Whenever you have to write an isolated piece of business logic, this pattern should be applicable.
+
+## Problem: 
+
+Usually, when you ask people where to place code, they all have their own opinion: on a table or page, or in some kind of codeunit library where lots of functions will be placed, or .. .  In fact, does it matter?  It sure does, because in many cases, the code simply isn't maintainable, let alone extendable or testable.
+
+## Description
+
+What if we have some kind of "standard way" to always write our code.  The *Generic Method Pattern* is kind of like what it says: a generic way to implement a method.
+
+### What is a method?
+
+Well, a method is *a significant piece of business logic* - may be best explained by some examples:
+
+- Posting a document
+- Any button on a page that executes business logic
+- a batch run to send emails
+- ...
+
+In a way, except "data validation", most of the things we write in our daily life, are methods.
+
+### The Pattern
+
+**One method, one codeunit**
+The idea is to put the code in one *encapsulated* codeunit with the purpose to have all the code in that one codeunit function for that one method.  This way, the codeunit will stay relatively small and readable.
+
+Let me start by showing an example, so you can re:
```

Comment:
Did you mean "so you can re" or is there missing something? Like "so you can re[late]" or similar? 🤔